packages/components/src/components/ResizableBox/ResizableBox.tsx

```diff
@@ -0,0 +1,209 @@
+    maxHeight?: number;
+};
+
+const reactiveAreaWidth = '15px';
+
+type ResizersProps = {
+    $width?: number;
+    $minWidth?: number;
+    $maxWidth?: number;
+    $height?: number;
+    $minHeight?: number;
+    $maxHeight?: number;
+    x?: number; // TODO not really used
+    y?: number;
+};
+
+const Resizers = styled.div<ResizersProps>(({ $width, $minWidth, $maxWidth, $height, $minHeight, $maxHeight, x, y }) => `
+        ${$width ? `width: ${$width}px;` : 'width: auto;'};
+        ${$minWidth && `min-width: ${$minWidth}px;`};
+        ${$maxWidth && `max-width: ${$maxWidth}px;`};
+        ${$height ? `height: ${$height}px;` : 'height: auto;'};
+        ${$minHeight && `min-height: ${$minHeight}px;`};
+        ${$maxHeight && `max-height: ${$maxHeight}px;`};
+        // ${x && `left: ${x}px;`};
+        // ${y && `top: ${y}px;`};
+        box-sizing: border-box;
+        position: relative;
+
+        #top-div, #left-div, #right-div, #bottom-div {
+          position: absolute;
+        }
+
+        #top-div, #bottom-div {
+          width: 100%;
+          height: ${reactiveAreaWidth};
+          cursor: ns-resize;
+        }
+
+        #left-div, #right-div {
+          width: ${reactiveAreaWidth};
+          height: 100%;
+          bottom: 0;
+          cursor: ew-resize;
+        }
+
+        #top-div {
+          top: ${`-${reactiveAreaWidth}`};
+        }
+
+        #left-div {
+          left: ${`-${reactiveAreaWidth}`};
+        }
+
+        #right-div {
+          right: ${`-${reactiveAreaWidth}`};
+        }
+
+        #bottom-div {
+          bottom: ${`-${reactiveAreaWidth}`};
+        }
+    `
+);
+
+const Child = styled(Resizers)`
+    position: relative;
+    width: 100%;
+    height: 100%;
+    overflow: scroll;
+`
+
+export const ResizableBox = ({
+    children,
+    directions,
+    isLocked = false,
+    width,
+    minWidth = 0,
+    maxWidth,
+    height,
+    minHeight = 0,
+    maxHeight,
+}: ResizableBoxProps) => {
+    const resizableBoxRef = useRef<HTMLDivElement>(null);
+
+    const [newX, setNewX] = useState<number | undefined>(undefined);
+    const [newY, setNewY] = useState<number | undefined>(undefined);
+    const [newWidth, setNewWidth] = useState<number>(width || minWidth || 0);
+    const [newHeight, setNewHeight] = useState<number>(height || minHeight || 0);
+
+    useEffect(() => {
+        // const resizers = Array.from(document.querySelectorAll(
+        //   directions.map(direction => `#${direction}-div`)
+        //   .join(', ')
+        // )) as Array<HTMLDivElement>;
+
+        // Note: adding event listers inside loop doesn't work
+        // resizers.forEach((resizer) => {
+        //   const myCallback = () => handleMouseDown(resizer.id);
+        //   resizer.addEventListener('mousedown', myCallback);
+        // });
+
+        directions.includes('bottom') && document.querySelector('#bottom-div')?.addEventListener('mousedown', () => handleMouseDown('bottom-div'));
```

Comment:
Please use conditions without returning value 🙏  (our eslint rules forbid using it this way)

```tsx
if(directions.includes('bottom')) {
  document.querySelector('#bottom-div')?.addEventListener('mousedown', () => handleMouseDown('bottom-div'));
}
```

etc.

Reply:
Oh, thanks, will fix that! :)